services/ui-src/src/utils/auth/UserProvider.tsx

```diff
@@ -36,12 +36,19 @@ export const UserProvider = ({ children }: Props) => {
   // state management
   const { user, showLocalLogins, setUser, setShowLocalLogins } = useStore();
 
+  // Clear selectedReport from localStorage when logging out of report.
+  const clearSelectedReportCache = () => {
+    const selectReportCacheExists = localStorage.getItem("selectedReportType");
+    selectReportCacheExists && localStorage.removeItem("selectedReportType");
+  };
+
   // initialize the authentication manager that oversees timeouts
   initAuthManager();
 
   const logout = useCallback(async () => {
     try {
       setUser(undefined);
+      clearSelectedReportCache();
       await Auth.signOut();
       localStorage.clear();
```

Comment:
There's a localStorage.clear() down here that should remove all the items from localStorage which is confusing to me of why you would need to do an individual remove?

Reply:
got an explanation from @keeysnc, i now understand why